core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -125,6 +125,10 @@ case class QueryPipelineWithFallback(queryChain: QueryChain
     val stats = new EngineQueryStats
     Try {
       val result = queryChain.execute(executorContext, rowListFn, QueryAttributes.empty, stats)
+      if(result.rowList.isEmpty){
```

Comment:
The aim is not to fallback for empty data, only for errors. There can be no rows for valid reasons and fallback is unnecessary.